### PR TITLE
Common backend interface

### PR DESCRIPTION
**Pull request by dmajere**
Good day,
I was thinking about common interface for all backends. This PR has an example of how it could be done, the main idea is to get dynamic backend loading without static lists in **init**.

### CODE REVIEW

File: testinfra/backend/ssh.py [hunk 1]
```diff
@@ -21,14 +21,14 @@
 from testinfra.backend import local
 
 
-class SshBackend(local.LocalBackend):
+class Backend(local.LocalBackend):
```
Comment (philpep) [inline testinfra/backend/ssh.py]: I think `local.LocalBackend` should be changed to `local.Backend`

File: testinfra/backend/__init__.py [hunk 2]
```diff
@@ -15,23 +15,13 @@
...
-))
+import importlib
 
 
 def get_backend(backend_type, *args, **kwargs):
     try:
-        backend_class = BACKENDS[backend_type]
-    except KeyError:
+        backend_module = importlib.import_module("testinfra.backend.%s" %
+                                                 backend_type)
```
Comment (philpep) [inline testinfra/backend/__init__.py]: I think the SafeSshBackend is broken here

**Comment by philpep** [comment]
Hi, thanks for your PR. I also think it's a better way to do that.

And this is related to #10 that I've just opened, maybe you can implement this feature in this PR too ?

**Comment by dmajere** [comment]
Hi, I'll try to do it during weekend.

**Comment by dmajere** [comment]
Added some code to cover #10. Added a small test for get_backend

**Comment by philpep** [comment]
Can one of the admins verify this patch?

**Comment by cw-kuoka** [comment]
@dmajere @philpep Is anyone working on this?

Though the code looks very promising to me, it has been several months since this PR created.
Today it has some conflicts but I guess I have some time to resolve them and recreate PR.

**Comment by mumoshu** [comment]
@dmajere @philpep Is anyone working on this?

Though the code looks very promising to me, it has been several months since this PR created.
Today it has some conflicts but I guess I have some time to resolve them and recreate PR.

**Comment by philpep** [comment]
@mumoshu note that this can currently be done by overriding the ``TestinfraBackend`` fixture in a ``conftest.py`` and implementing a ``testinfra.backend.base.BaseBackend`` like class.

I'm not sure yet about implementing this feature like this PR does, I must be more convinced. What custom backend would you like to use ? Maybe I could try to implement it outside of testinfra, this will give me an idea of required changes (or not) in testinfra.

(sorry for the late response)

**Pull request closed without merging**